common/script/index.coffee

```diff
@@ -1740,14 +1744,22 @@ api.wrap = (user, main=true) ->
           dailyChecked += 1
         else
           # dailys repeat, so need to calculate how many they've missed according to their own schedule
+          console.log("task: " + task.text) # XXX Alys to remove these
           scheduleMisses = 0
+          keepGoing = true # XXX Alys to improve how we break out of _.times
           _.times daysMissed, (n) ->
```

Comment:
We do need to use daysMissed as the parameter to _.times. We have to check back through each day missed until we find a day on which the Daily was due, otherwise we can't correctly untick any ticked checklist items, and we can't assign damage for one day missed. We just need to break out of _.times as soon as we've found one day that was missed.